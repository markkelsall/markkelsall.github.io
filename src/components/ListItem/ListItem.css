.ui-autocomplete li a { 
    color: #222222;
    border-radius: 2px;
    border-bottom: 1px solid #eaeaea;
    padding: 10px 9px;
    display: block;
    display: flex;
    flex-direction: row;
}

.ui-autocomplete li :hover { 
    background-color: #e6f1f9;
}

.ui-corner-all {
    border-radius: 4px;
    -webkit-border-radius: 4px;
}

.ui-corner-all a span {
    min-width: 50px;
    height: 15px;
    margin: 2px 10px 0 0;
    padding: 3px 5px;
    border-radius: 3px;
    text-align: center;
    font-weight: bold;
    font-size: 85%;
}

.airport {
    background-color: #f2817f;
}

.city {
    background-color: #6eb8fa;
}

.district {
    background-color: #70d58e;
}

.type {
    color: #fff;
}

.location {
    display: flex;
    flex-direction: column;
}

.address {
    margin: 0 !important;
    padding: 0 !important;
    font-size: 85%;
    font-weight: unset !important;
    color: gray;
    font-style: normal;
}