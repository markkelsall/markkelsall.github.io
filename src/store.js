import {createStore, applyMiddleware, combineReducers} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import pickupReducer from './reducers/pickUpReducer';

const reducer = combineReducers({
  pickUp: pickupReducer
})

export default createStore(
  reducer,
  composeWithDevTools(
    applyMiddleware(thunk)
  )
)