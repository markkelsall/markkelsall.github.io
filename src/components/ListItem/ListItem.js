import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './ListItem.css';

export default class ListItem extends Component {

    static propTypes = {
        item: PropTypes.object,
        searchTerm: PropTypes.string
    };

    render = () => {
        let type;
        switch(this.props.item.location.placeType) {
        case "A":
            type = 'airport';
            break;
        case "C":
            type = 'city';
            break;
        case "D":
            type = 'district';
            break;
        default:
            type = "A place";
        }

        return (
            <li className="ui-menu-item" role="menuitem" tabIndex="-1">
                <a className="ui-corner-all">
                    <span className={`${type} type`}>{type}</span>
                    <div className="location">
                        <em>{this.props.item.place}</em>
                        <span className="address">{this.props.item.address['1']}</span>
                    </div>
                </a>
            </li>
        );
    }
}
