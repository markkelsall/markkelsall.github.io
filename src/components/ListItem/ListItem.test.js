import React from 'react'
import renderer from 'react-test-renderer'
import ListItem from './ListItem'
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from '../../store'

it ('Should render properly', () => {
    const item = {
        location: {
            placeType: 'A'
        },
        address: {
            '1': 'Street'
        }
    };
    const component = renderer.create(
        <Provider store={store}><Router><ListItem item={item}/></Router></Provider>,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})