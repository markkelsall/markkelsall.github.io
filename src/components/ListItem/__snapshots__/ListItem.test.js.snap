// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render properly 1`] = `
<li
  className="ui-menu-item"
  role="menuitem"
  tabIndex="-1"
>
  <a
    className="ui-corner-all"
  >
    <span
      className="airport type"
    >
      airport
    </span>
    <div
      className="location"
    >
      <em />
      <span
        className="address"
      >
        Street
      </span>
    </div>
  </a>
</li>
`;
